templates/zoo.cfg.j2

```diff
@@ -2,14 +2,20 @@ tickTime={{ tick_time }}
 dataDir={{ data_dir }}
 dataLogDir={{ log_dir }}
 clientPort={{ client_port }}
+
+{# More than 1 host ==> Quorum #}
+
+{% if zookeeper_hosts|length > 1 %}
 initLimit={{ init_limit }}
 syncLimit={{ sync_limit }}
+{% endif %}
 
-{#
-Also consider:
-server.{{server.id}}={{server.host}}:2888:3888
-#}
-
+{% if zookeeper_hosts is sequence %}
```

Comment:
If I got the intention right, I think the logic can actually simplified by simply testing if the list item has a property `host`, in which case we use it otherwise we use the item itself, assuming it is the host name. Then no need to test for sequence or mapping.
```
{% for server in zookeeper_hosts %}
server.{{loop.index}}={{ server.host if server.host is defined else server }}:2888:3888
{% endfor %}
```